### ISSUE TITLE
Avoidable build failures and warnings

### ISSUE THREAD

**Issue by eddelbuettel** [r-hub/rhub#49]
Briefly:
- Note regarding 'Cannot check README.md for lack of pandoc' when pandoc is just and apt-get away, either from Ubuntu or from the upstream site
- RProtoBuf failing on Ubuntu 16.04 -- you didn't get the sys.deps rights, this is _easily_ buildable (and my development platform)
- RProtoBuf failing on Windows -- builds fine on win-builder so could be buildable
- PRrotoBuf failing on Fedora for lack of fonts for vignette, maybe my issue for lack of `--no-vignettes` argument?

I would also be _really nice_ if we could see `00install.out` in case of failure.

**Comment by gaborcsardi** [comment]
Thanks!

> - Note regarding 'Cannot check README.md for lack of pandoc' when pandoc is just and apt-get away, either from Ubuntu or from the upstream site

You can specify pandoc in `SystemRequirements`, although this might not be the best solution, and I guess I should just install it on all images by default. It is easy on Debian, but painful on many others....

> - RProtoBuf failing on Ubuntu 16.04 -- you didn't get the sys.deps rights, this is _easily_ buildable (and my development platform)

Should be OK now.

> - RProtoBuf failing on Windows -- builds fine on win-builder so could be buildable

Yes, I'll take a look in a minute.

> - PRrotoBuf failing on Fedora for lack of fonts for vignette, maybe my issue for lack of `--no-vignettes` argument?

Which fonts are missing?

> I would also be _really nice_ if we could see `00install.out` in case of failure.

Indeed. I am working on artifacts today.

**Comment by eddelbuettel** [comment]
> Which fonts are missing?

Don't recall the name but maybe you can mock--many of my vignettes (and in this case, `RProtoBuf-intro.Rnw`) have been using these three lines for maybe half a decade:

```latex
\usepackage{microtype}                     %% cf http://www.khirevich.com/latex/microtype/
\usepackage[T1]{fontenc}		   %% cf http://www.khirevich.com/latex/font/
\usepackage[bitstream-charter]{mathdesign} %% cf http://www.khirevich.com/latex/font/
```

Thanks for all the other fixes.  And yes, `pandoc` should be omnipresent now that `README.md` is a given.

**Comment by gaborcsardi** [comment]
OK, Linux images should have pandoc now, some reasonable version. Still not good enough for everything on Ubuntu and Fedora (https://github.com/r-hub/pandoc-builders/issues/2, https://github.com/r-hub/pandoc-builders/issues/1), but I'll deal with that later.

TODO: 
- [ ] RProtoBuf on windows
- [ ] sysreq for microtype font

**Comment by eddelbuettel** [comment]
BTW I could understand if you drew a line and said "xyz won't be supported on platform abc".  It is hard to do _all_ things to _all_ people.  Eg the new-ish sf package required libgdal versions not present.  I may just blacklist that for my (own, internal) Rcpp rev.dep tests.

**Comment by gaborcsardi** [comment]
The plan is to have "primary" platforms, and "secondary" ones with different levels of support. E.g. I'll only build binary CRAN packages for primary ones. Also, I'll only update images/VMs on a schedule, (say) weekly, once they stabilize....

**Comment by gaborcsardi** [comment]
`install.out` is now available with other artifacts as well: http://log.r-hub.io/2016/11/22/2016-11-22-artifacts/

**Comment by gaborcsardi** [comment]
> Note regarding 'Cannot check README.md for lack of pandoc' when pandoc is just and apt-get away, either from Ubuntu or from the upstream site

I'll check that pandoc is available everywhere (except for Solaris which is not possible imo): #113 

> RProtoBuf failing on Ubuntu 16.04 -- you didn't get the sys.deps rights, this is easily buildable (and my development platform)

https://sysreqs.r-hub.io/pkg/RProtoBuf
This should be ok, no? 

> RProtoBuf failing on Windows -- builds fine on win-builder so could be buildable

Maybe fixed already, but will check: #114

> PRrotoBuf failing on Fedora for lack of fonts for vignette, maybe my issue for lack of --no-vignettes argument?

You can supply check arguments: `rhub::check(check_args = "--no-vignettes")` But also #115
